base/math.jl

```diff
@@ -166,6 +170,43 @@ function hypot{T<:AbstractFloat}(x::T, y::T)
     x * sqrt(one(r)+r*r)
 end
 
+# 3-arg hypot
+hypot(x::Real, y::Real, z::Real) =
+    hypot(promote(float(x), float(y), float(z))...)
+function hypot{T<:AbstractFloat}(x::T, y::T, z::T)
+    x = abs(x)
+    y = abs(y)
+    z = abs(z)
+    m = max(x, y, z)
+    if isinf(m)
+        return m
+    elseif isfinite(3*m*m) && m*m != 0 # Scaling not necessary
+        return sqrt(abs2(x) + abs2(y) + abs2(z))
+    elseif m == 0
+        o = one(m)
+        return m*sqrt(abs2(x/o) + abs2(y/o) + abs2(z/o))
+    else
+        return m*sqrt(abs2(x/m) + abs2(y/m) + abs2(z/m))
+    end
+end
+
+# varargs hypot
+hypot(w::Real, x::Real, y::Real, z::Real...) =
+    hypot(promote(float(w), float(x), float(y), float([z...])...)...)
+function hypot{T<:AbstractFloat}(w::T, x::T, y::T, z::T...)
+    a = abs([w, x, y, z...])
```

Comment:
> Creating an array is very expensive, compared to the `hypot` function. If that's the only way to implement this, then I'd rather not have this functionality, as this looks like a performance trap (i.e. a bad surprise).

Thanks for your comment.  I don't know if it's the only way to implement this.

> A `hypot` function that takes an array (or a collection, or an iterable) as argument might make sense, though.

But that would be a complete duplicate of `vecnorm`, right?